### DISCUSSION TITLE
How to configure fast-xml-parser for consistent array outputs?

### DISCUSSION THREAD

**Discussion by patjakubik** [NaturalIntelligence/fast-xml-parser#660] [Q&A]
Hi there,
I've been using FXP for quite some time and I love it, thank you for all the work that you put in.

I started struggling with the arrays, especially in situations where array size is 0 or 1.

If it is 0 - it is parsed as an empty string.

![Screenshot 2024-06-17 at 23 09 29@2x](https://github.com/NaturalIntelligence/fast-xml-parser/assets/37963339/530ce15e-0413-4882-8ee4-89c8051e917c)

If it is 1 - it is parsed as an object.

![Screenshot 2024-06-17 at 23 09 20@2x](https://github.com/NaturalIntelligence/fast-xml-parser/assets/37963339/2f41344c-563b-4985-8a98-fe60ad7f4d35)

If it is more than 1 - it is parsed as an array

![Screenshot 2024-06-17 at 23 09 05@2x](https://github.com/NaturalIntelligence/fast-xml-parser/assets/37963339/715cec05-2f0d-4245-a48d-e2134677858c)

Would it be possible to pass `any_name` as an option to tell the parser that this jPath should be always parsed as array?

**Comment by amitguptagwl** [comment]
I'm assuming that you must be using v4 of the parser. Have you tries `isArray` and `updateTag` options? If possible, try v5 as well. Please check README for more detail.

**Reply by patjakubik** [reply]
Hi @amitguptagwl 

I indeed tried the `isArray` option but haven't noticed any difference in output when returning true vs false. And `updateTag` seems to be changing the name so this is not really what I need. I am on 4.3.5. Will try v5, which it seems to be bundled with 4.x releases right?

**Reply by amitguptagwl** [reply]
yes v5 is bundled with v4.

**Comment by amitguptagwl** [comment]
@patjakubik I've launched another XML Parser yesterday that should fulfill your requirement.

[@nodable/flexible-xml-parser](https://github.com/nodable/flexible-xml-parser)
